### PR TITLE
  Add a prompt when a user tries to leave a page with a document that has changes without saving

### PR DESCRIPTION
**Pull request by usrtim**
Add a prompt when a user tries to leave a page with a document that has changes without saving

### CODE REVIEW

File: translations/en-us.yaml [hunk 1]
```diff
@@ -100,3 +100,5 @@ reglementaireBijlageTitel:
   description: "Regulatory statement title"
   placeholder: "Untitled document"
   modify: "Modify title"
+behandelingVanAgendapunten:
+  confirmLeaveWithoutSaving: Bent u zeker dat u deze pagina wilt verlaten zonder uw wijzigingen op te slaan?
```
Comment (abeforgit) [inline translations/en-us.yaml:104]: change to
"Are you sure you want to leave this page without saving your changes?"

also, the key "behandelingVanAgendapuntent" (treatment of an agenda item)  doesn't make sense in this project. I'd just put it under reglementEdit (make sure to do that both in this file and in the dutch file)

**Review by abeforgit — CHANGES_REQUESTED** [review]

File: translations/en-us.yaml (hunk 1 above)
Comment (usrtim) [reply translations/en-us.yaml:104]: ok, sorry but didn't understand that :D

**Review by abeforgit — APPROVED** [review]

File: translations/en-us.yaml (hunk 1 above)
Comment (abeforgit) [reply translations/en-us.yaml]: yeah no worries you couldnt have known

**Review by abeforgit — CHANGES_REQUESTED** [review]
I approved this too quickly

when you visit a document, make changes, and then leave without saving, the prompt doesn't appear

**Comment by usrtim** [comment]
> I approved this too quickly
> 
> when you visit a document, make changes, and then leave without saving, the prompt doesn't appear
> 
> EDIT: that's not quite accurate
> 
> here's what's happening:
> 
>     * load the list page from scratch (aka hit F5)
> 
>     * visit document, make changes, leave without saving
>       -> the prompt doesn't appear
> 
> 
> now, save any document once
> 
>     * visit document make changes, leave without saving
>       -> the prompt appears correctly
> 
> 
> BUT
> 
>     * visit any document, don't make any changes at all, and leave
>       -> the prompt appears anyway!

thanks @abeforgit nice catch. Now tested and it's working as expected. However, when I go from a document that I just created, to some older documents prompt appears anyway!. What I found It's that structure it's a bit different(as you can see in image below). Based on my discussion with @elpoelma , the reason why the structure is a bit different, is because of some changes the inline-component API has gone through: the attributes are somewhat modeled different.
But after saving a first time, these changes should no longer appear as the document has been updated to the new API.

Screenshot 2022-12-02 at 3 59 15 PM

**Comment by usrtim** [comment]
> still not right I'm afraid
> 
>     * go to an RS
> 
>     * make a change
> 
>     * click the browser back button
> 
>     * click cancel on the popup
> 
>     * click the browser back button again
> 
>     * click ok
>       -> you stay on the edit page
> 
> 
> still a problem the transition cancellation logic

Even I couldn't reproduce this case, I saw the same problem that we had on **frontend-gelinkt-notuleren** with browser back button and fix that.

**Pull request merged**